### PR TITLE
Cleanup of query related tests. Fixed usage of wrong test method.

### PR DESCRIPTION
**Pull request by Donnerbart**
* Cleanup of query related tests.
* Fixed a copy & paste bug which was hidden by not using Generics in tests.

I was actually searching for a test to extend to help out for missing code coverage of query related features. To not mix up cleanups and changes I made this PR. Nothing exiting here, mostly driven by IntelliJ warnings.

The only behavior change should be the found copy & paste bug, which was uncovered by adding Generics to the test maps. I'll make a line comment to point out the fix.

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/map/impl/query/QueryBasicTest.java [hunk 1]
```diff
@@ -401,19 +401,19 @@ private void testPredicateDateAttribute(IMap map) {
...
     }
 
     @Test(timeout = 1000 * 60)
     public void testPredicateEnumAttributeWithIndex() {
         HazelcastInstance instance = createHazelcastInstance(getConfig());
-        IMap map = instance.getMap("testPredicateEnumAttribute");
+        IMap<Integer, NodeType> map = instance.getMap("testPredicateEnumAttribute");
         map.addIndex("this", true);
-        testPredicateDateAttribute(map);
+        testPredicateEnumAttribute(map);
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/map/impl/query/QueryBasicTest.java]: Here is the fixed method call: `testPredicateDateAttribute()` vs. `testPredicateEnumAttribute()`.

**Comment by Donnerbart** [comment]
verify

**Comment by Donnerbart** [comment]
verify

**Comment by Donnerbart** [comment]
verify

**Comment by Donnerbart** [comment]
run-lab-run

**Comment by mtopolnik** [comment]
:+1:

**Comment by Donnerbart** [comment]
Since the changes are many, but very trivial and just in test code, I'll merge this to be able to continue with my original task (increase missing code coverage in query related features).

**Pull request merged**